### DISCUSSION TITLE
Where are the best places to host dynamic websites ?

### DISCUSSION THREAD

**Discussion by jbampton** [brisbanesocialchess/awesome-social-chess#83] [General]
Looking for opinions on the best places to host websites.

Things to consider:

- cost
- ease of use
- integration with GitHub
- which website framework to use
- should we use serverless 

Starting with a static website but in future need a database or some kind of dynamic website / storage of data.

Some places that I know about:

- [Heroku](https://www.heroku.com)
- [Cloudflare Workers](https://workers.cloudflare.com/)
- [Digital Ocean](https://www.digitalocean.com/)
- [Linode](https://www.linode.com/)
- some Linux based servers roll your own etc
- others long forgotten :)

For just a static website

- [Cloudflare Pages](https://pages.cloudflare.com/)
- [GitHub Pages](https://pages.github.com/)

**Comment by MohammadLotfiA** [comment]
Digital Ocean is okay, blue host as well.

If it's a static website, for now we can also use the GitHub server too.

About the platform, I think vanilla JS + HTML + CSS is gonna do.

**Comment by aniruddhaadak80** [comment]
Hey @jbampton ,

For hosting dynamic websites, consider the following options based on your criteria:


1. **Vercel**:
- **Cost**: Free for personal projects; pay-as-you-go for teams. [Vercel Pricing](https://vercel.com/pricing)
- **Ease of Use**: Optimized for frontend as well as backend frameworks  . 
- **Integration with GitHub**: Excellent integration for deployments. [Vercel GitHub Integration](https://vercel.com/docs/git)

2. **Netlify**:
- **Cost**: Free tier available; paid plans for larger teams. [Netlify Pricing](https://www.netlify.com/pricing/)
- **Ease of Use**: Very user-friendly for static and dynamic sites. 
- **Integration with GitHub**: Simple to connect for CI/CD. 

Both are very flexible and easy to use .

Thanks .

**Comment by BaseMax** [comment]
All on git and fully free
github pages + jekyll/yml

**Comment by AmirTallap** [comment]
For a starter static website, github pages are OK, but you will have to consider the cost while you move forward with dynamic content, things get out of control pretty quickly, especially cost wise. There are a tonne of tools that will need monthly or per query subscription.
